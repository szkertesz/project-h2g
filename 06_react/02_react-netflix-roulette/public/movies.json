{
  "movies": [
    {
      "id": "8a3ec45b-b0a4-4ebd-a56e-ef618f40dffc",
      "title": "Martyrs",
      "release_date": "2008",
      "genre": "Horror",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BYTY0NTZlZjYtNWRmZi00MGRjLTk4ZDctMjE4NjBiZTUyNGNhXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_.jpg",
      "movie_url": "https://www.imdb.com/title/tt1029234",
      "rating": "7.0",
      "runtime": "99 min",
      "description": "A young woman's quest for revenge against the people who kidnapped and tormented her as a child leads her and a friend, who is also a victim of child abuse, on a terrifying journey into a living hell of depravity."
    },
    {
      "id": "c756ce95-7ca8-4bbd-8968-8296030",
      "title": "You Won't Be Alone",
      "release_date": "2022",
      "genre": "Horror, Drama",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BMmU2MDkyZWMtYjJmNi00YjE1LWIzODctZjNjZmY0OGMzNzQzXkEyXkFqcGdeQXVyMDM2NDM2MQ@@._V1_.jpg",
      "movie_url": "https://www.imdb.com/title/tt8296030/",
      "rating": "6.3",
      "runtime": "108 min",
      "description": "Set in an isolated mountain village in 19th century Macedonia, You Won't Be Alone follows a young girl who is kidnapped and then transformed into a witch by an ancient spirit. Curious about life as a human, the young witch accidentally kills a peasant in the nearby village and then takes her victim's shape to live life in her skin. Her curiosity ignited, she continues to wield this horrific power in order to understand what it means to be human."
    },
    {
      "id": "9951d425-6f87-4f76-b6cf-063fc15fe2a0",
      "title": "Stranger Things",
      "release_date": "2016",
      "genre": "Drama, Fantasy, Horror",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BODZlYjQ4NzYtZTg1MC00NGY4LTg4NjQtNGE3ZjRkMjk3YjMyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_SX300.jpg",
      "movie_url": "https://www.imdb.com/title/tt4574334/",
      "rating": "8.7",
      "runtime": "51 min",
      "description": "In a small town where everyone knows everyone, a peculiar incident starts a chain of events that leads to a child's disappearance, which begins to tear at the fabric of an otherwise-peaceful community. Dark government agencies and seemingly malevolent supernatural forces converge on the town, while a few of the locals begin to understand that more is going on than meets the eye."
    },
    {
      "id": "856aae4a-3593-4571-adc9-d95540498383",
      "title": "Star Wars",
      "release_date": "1977",
      "genre": "Action, Adventure, Fantasy",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BNzg4MjQxNTQtZmI5My00YjMwLWJlMjUtMmJlY2U2ZWFlNzY1XkEyXkFqcGdeQXVyODk4OTc3MTY@._V1_SX300.jpg",
      "movie_url": "https://www.imdb.com/star-wars/",
      "rating": "8.6",
      "runtime": "121 min",
      "description": "Luke Skywalker joins forces with a Jedi Knight, a cocky pilot, a Wookiee and two droids to save the galaxy from the Empire's world-destroying battle station, while also attempting to rescue Princess Leia from the mysterious Darth Vader."
    },
    {
      "id": "318119d2-67ea-4e98-a606-e0916f0bf3dc",
      "title": "Thor",
      "release_date": "2011",
      "genre": "Action, Adventure, Fantasy",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BOGE4NzU1YTAtNzA3Mi00ZTA2LTg2YmYtMDJmMThiMjlkYjg2XkEyXkFqcGdeQXVyNTgzMDMzMTg@._V1_SX300.jpg",
      "movie_url": "https://www.imdb.com/title/tt0800369/",
      "rating": "7.0",
      "runtime": "115 min",
      "description": "The warrior Thor (Chris Hemsworth) is cast out of the fantastic realm of Asgard by his father Odin (Sir Anthony Hopkins) for his arrogance and sent to Earth to live amongst humans. Falling in love with scientist Jane Foster (Natalie Portman) teaches Thor much-needed lessons, and his new-found strength comes into play as a villain from his homeland sends dark forces toward Earth."
    },
    {
      "id": "e67fb515-0b36-4f9d-a7fa-aa38150b66bb",
      "title": "Pulp Fiction",
      "release_date": "1994",
      "genre": "Crime, Drama",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
      "movie_url": "https://www.imdb.com/title/tt0110912/",
      "rating": "8.9",
      "runtime": "154 min",
      "description": "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents."
    },
    {
      "genre": "documentary,horror",
      "url": "dsaf",
      "rating": "1",
      "release": "2022-07-08",
      "runtime": "4323",
      "thumbnail": "",
      "title": "sadf",
      "description": "loremview",
      "id": "KUVGi4y"
    },
    {
      "id": "8LNL6zGFMwKMuVnSF8XvZ",
      "genre": "crime,horror",
      "url": "sadffdsa",
      "rating": "1.2",
      "release": "2022-07-01",
      "runtime": "213",
      "thumbnail": "",
      "title": "sadffdsa",
      "description": "1345bvcx"
    },
    {
      "id": "d48fucm4WXXzFYvIp5xe4",
      "genre": "horror",
      "url": "https://www.imdb.com/title/tt4954522/?ref_=tt_mv_close",
      "rating": "8",
      "release": "2019-06-12",
      "runtime": "311",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BMTU3MDUxMDI0MV5BMl5BanBnXkFtZTgwMzk3OTg3MDI@._V1_.jpg",
      "title": "Raw",
      "description": "A young woman, studying to be a vet, develops a craving for human flesh."
    },
    {
      "id": "LNeN9WumyhIb6VF72wgc3",
      "genre": "horror",
      "url": "https://m.media-amazon.com/images/M/MV5BNTAyOTkxNTQ3NV5BMl5BanBnXkFtZTgwMjQ1NzQxOTE@._V1_.jpg",
      "rating": "7",
      "release": "",
      "runtime": "150",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BNTAyOTkxNTQ3NV5BMl5BanBnXkFtZTgwMjQ1NzQxOTE@._V1_.jpg",
      "title": "Film",
      "description": "An aspiring model, Jesse, is new to Los Angeles. However, her beauty and youth, which generate intense fascination and jealousy within the fashion industry, may prove themselves sinister.\n"
    },
    {
      "id": "NLJn_hJIKxt9rs-0vJsu2",
      "genre": "horror,comedy",
      "url": "vb",
      "rating": "1.1",
      "release": "2022",
      "runtime": "342",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BMTU3MDUxMDI0MV5BMl5BanBnXkFtZTgwMzk3OTg3MDI@._V1_.jpg",
      "title": "sadf",
      "description": "gsfd"
    },
    {
      "id": "rfKAmFIPPxkbsfePLpDCT",
      "genre": "documentary",
      "url": "https://www.imdb.com/title/tt4954522/?ref_=tt_mv_close",
      "rating": "3.6",
      "release_date": "2020",
      "runtime": "213",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BMTU3MDUxMDI0MV5BMl5BanBnXkFtZTgwMzk3OTg3MDI@._V1_.jpg",
      "title": "fdsa",
      "description": "An aspiring model, Jesse, is new to Los Angeles. However, her beauty and youth, which generate intense fascination and jealousy within the fashion industry, may prove themselves sinister.\n"
    },
    {
      "id": "08Rtz0n7jEpfnNiSmpwGC",
      "genre": "documentary,horror",
      "url": "https://www.imdb.com/title/tt4954522/?ref_=tt_mv_close",
      "rating": "1",
      "release_date": "2021",
      "runtime": "123",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BNTAyOTkxNTQ3NV5BMl5BanBnXkFtZTgwMjQ1NzQxOTE@._V1_.jpg",
      "title": "sdf",
      "description": "An aspiring model, Jesse, is new to Los Angeles. However, her beauty and youth, which generate intense fascination and jealousy within the fashion industry, may prove themselves sinister.\n"
    },
    {
      "id": "nmUTn4VoIZOYhdI-td4ru",
      "genre": "comedy",
      "url": "fdsa",
      "rating": "10",
      "release_date": "2022",
      "runtime": "123",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BMTU3MDUxMDI0MV5BMl5BanBnXkFtZTgwMzk3OTg3MDI@._V1_.jpg",
      "title": "newest film",
      "description": "An aspiring model, Jesse, is new to Los Angeles. However, her beauty and youth, which generate intense fascination and jealousy within the fashion industry, may prove themselves sinister.\n"
    },
    {
      "id": "Ycdr7LAYJSrz46PtQi03s",
      "genre": "horror",
      "url": "https://www.imdb.com/title/tt1974419/?ref_=tt_mv_close",
      "rating": "6",
      "release_date": "2013",
      "runtime": "123",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BMTU3MDUxMDI0MV5BMl5BanBnXkFtZTgwMzk3OTg3MDI@._V1_.jpg",
      "title": "Raw",
      "description": "https://www.imdb.com/title/tt1974419/?ref_=tt_mv_close"
    },
    {
      "id": "hV3Z779-8QW057R8OkWMU",
      "genre": "crime",
      "url": "https://www.imdb.com/title/tt4954522/?ref_=tt_mv_close",
      "rating": "3",
      "release_date": "2022",
      "runtime": "234",
      "thumbnail": "https://m.media-amazon.com/images/M/MV5BNTAyOTkxNTQ3NV5BMl5BanBnXkFtZTgwMjQ1NzQxOTE@._V1_.jpg",
      "title": "fetching",
      "description": "sadf"
    }
  ]
}