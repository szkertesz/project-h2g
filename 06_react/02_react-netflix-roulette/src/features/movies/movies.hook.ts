import { useAppSelector } from '../../app/hooks';
import { Movie } from '../../interfaces/Movie';

export const moviesSelector = (state: any) => {
    console.log(state)
    return state.movies.data
        .filter((movie: Movie) =>
            movie.title
                .toLowerCase()
                .includes(state.movies.searchedMovie.toLowerCase())
        )
        .filter(
            (movie: Movie) =>
                state.movies.filterOptions.genre === '' ||
                movie.genres
                    .map((genre) => genre.toLowerCase())
                    .includes(state.movies.filterOptions.genre)
        )
        .sort((a: any, b: any) => {
            if (state.movies.sortOptions.sortOrder === 'ascending') {
                switch (state.movies.sortOptions.sortCriterion) {
                    case 'releaseDate':
                        return (
                            new Date(a.release_date).getTime() -
                            new Date(b.release_date).getTime()
                        );
                    case 'runtime':
                        return (
                            a.runtime - b.runtime
                        );
                    case 'rating':
                        return a.vote_average - b.vote_average;
                }
            } else {
                switch (state.movies.sortOptions.sortCriterion) {
                    case 'releaseDate':
                        return (
                            new Date(b.release_date).getTime() -
                            new Date(a.release_date).getTime()
                        );
                    case 'length':
                        return (
                            b.runtime - a.runtime
                        );
                    case 'rating':
                        return b.vote_average - a.vote_average;
                }
            }
            return a.release_date - b.release_date;
        });
};

export const useMovies = () => useAppSelector(moviesSelector);
