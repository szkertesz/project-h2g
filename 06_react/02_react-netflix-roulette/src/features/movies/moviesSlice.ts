import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { client } from '../../services/client'
import { Movie } from '../../interfaces/Movie';
import type { RootState } from '../../app/store';

interface moviesState {
    data: Movie[],
    status: 'idle' | 'loading' | 'succeeded' | 'failed',
    error: string | null | undefined,
    searchedMovie: string;
    sortOptions: {
        sortCriterion: string;
        sortOrder: string;
    };
    filterOptions: {
        genre: string;
    };
}

interface MovieToAddWithId extends Movie {
    tagline: string,
    vote_count: number,
    budget: number
};

type MovieToAdd = Omit<MovieToAddWithId, 'id'>

const initialState: moviesState = {
    data: [],
    status: 'idle',
    error: null,
    searchedMovie: '',
    sortOptions: {
        sortCriterion: '',
        sortOrder: '',
    },
    filterOptions: {
        genre: '',
    },
};

export const fetchMovies = createAsyncThunk('movies/fetchMovies', async () => {
    const response = await client.get('http://localhost:4000/movies?limit=50');
    return response.data.data;
});
export const addMovie = createAsyncThunk(
    'movies/addMovie',
    async (newMovieData: MovieToAdd, { rejectWithValue }) => {
        try {
            const response = await client.post(
                'http://localhost:4000/movies',
                newMovieData
                );
                return response.data;
        } catch (error) {
            return rejectWithValue(error)
        }
    }
);
export const editMovie = createAsyncThunk(
    'movies/editMovie',
    async (newMovieData: Movie, { rejectWithValue }) => {
    try {
        const response = await client.put(
            `http://localhost:4000/movies`,
            newMovieData
        );
        return response.data;
    } catch (error) {
        return rejectWithValue(error);
    }
});
// export const filterMoviesByGenre = createAsyncThunk(
//     'movies/filterMoviesByGenre',
//     async (genres: string, thunkAPI) => {
//         const response = await client.get(
//             `http://localhost:4000/movies?filter=${genres}`
//         );
//         return response.data.data;
//     }
// );

// export const sortMoviesByCriterion = createAsyncThunk<Movie[], {selectedCriterion: string, direction: string}>(
//     'movies/sortMoviesByCriterion',
//     async ({selectedCriterion, direction}) => {
//         const response = await client.get(
//             `http://localhost:4000/movies?sortBy=${selectedCriterion}&sortOrder=${direction}`
//         );
//         return response.data.data;
//     }
// );
const moviesSlice = createSlice({
    name: 'movies',
    initialState,
    reducers: {
        // editMovie(state, action) {
        //     const {
        //         id,
        //         title,
        //         genres,
        //         vote_average,
        //         release_date,
        //         runtime,
        //         poster_path,
        //         overview,
        //     } = action.payload;
        //     const movieToEdit = state.data.find((movie) => movie.id === id);

        //     if (movieToEdit) {
        //         movieToEdit.id = id;
        //         movieToEdit.title = title;
        //         movieToEdit.genres = genres;
        //         movieToEdit.vote_average = vote_average;
        //         movieToEdit.release_date = release_date;
        //         movieToEdit.runtime = runtime;
        //         movieToEdit.poster_path = poster_path;
        //         movieToEdit.overview = overview;
        //     }
        // },
        setSearchedMovie: (state, action) => {
            state.searchedMovie = action.payload;
        },
        setGenreFilter: (state, action) => {
            state.filterOptions.genre = action.payload;
        },
        setSortParams: (state, action) => {
            state.sortOptions.sortCriterion = action.payload.criterion;
            state.sortOptions.sortOrder = action.payload.order;
        },
    },
    extraReducers(builder) {
        builder
            .addCase(fetchMovies.pending, (state, action) => {
                state.status = 'loading';
            })
            .addCase(fetchMovies.fulfilled, (state, action) => {
                state.status = 'succeeded';
                // Add any fetched movies to the array
                state.data = state.data.concat(action.payload);
            })
            .addCase(fetchMovies.rejected, (state, action) => {
                state.status = 'failed';
                state.error = action.error.message;
            })
            .addCase(addMovie.fulfilled, (state, action) => {
                state.data.push(action.payload);
            })
            .addCase(editMovie.fulfilled, (state, action) => {
                console.warn(action.payload);
                const index = state.data.findIndex(
                    (movie) => movie.id === action.payload.id
                );
                state.data[index] = action.payload;
            });
        // .addCase(filterMoviesByGenre.fulfilled, (state, action) => {
        //     state.data = action.payload
        // })
        // .addCase(sortMoviesByCriterion.fulfilled, (state, action) => {
        //     state.data = action.payload
        // })
    },
});

// export const { editMovie } = moviesSlice.actions
export const { setSearchedMovie, setGenreFilter, setSortParams } = moviesSlice.actions;

export default moviesSlice.reducer

// Selectors
export const selectAllMovies = (state: RootState) => state.movies.data;

export const selectMovieById = (state: RootState, movieId: number) =>
    state.movies.data.find((movie) => movie.id === movieId);