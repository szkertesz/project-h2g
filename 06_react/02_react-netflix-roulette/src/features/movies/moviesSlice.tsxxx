import { createSlice } from '@reduxjs/toolkit';
const initialState = [
    { id: '1', title: 'First Post!', content: 'Hello!' },
    { id: '2', title: 'Second Post', content: 'More text' },
];
const moviesSlice = createSlice({
    name: 'movies',
    initialState,
    reducers: {
        movieAdded(state, action) { // the action creator
            state.push(action.payload)
        },
        movieUpdated(state, action) {
            const {id, genre, url, rating, release, runtime, thumbnail, title, description} = action.payload
            const existingMovie = state.find(movie => movie.id === id)
            if (existingMovie) {
                existingMovie.genre = genre
                existingMovie.url = url
                existingMovie.rating = rating
                existingMovie.release = release
                existingMovie.runtime = runtime
                existingMovie.thumbnail = thumbnail
                existingMovie.title = title
                existingMovie.description = description
            }
        }
    }
})

export const { movieAdded } = moviesSlice.actions

export default moviesSlice.reducer