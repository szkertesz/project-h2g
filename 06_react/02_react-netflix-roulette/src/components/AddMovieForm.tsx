import { useAppDispatch } from '../app/hooks';
import { addMovie } from '../features/movies/moviesSlice';
import { SetStateAction, useState } from 'react';

import classes from './AddMovieForm.module.scss';
import modalClasses from './Modal.module.scss';
import Button from '../ui/Button';
import CustomSelect from './CustomSelect';
import Modal from '../components/Modal';
import IconCheck from '../assets/images/icon-check.svg';
// RTKQ code
// import { useAddNewDataMutation } from '../features/api/apiSlice';

interface Props {}

type Genres = string[];

const AddMovieForm: React.FC<Props> = () => {
    const [genres, setGenre] = useState<Genres>([]);
    const [vote_average, setRating] = useState(1);
    const [release_date, setRelease] = useState('');
    const [runtime, setRuntime] = useState(1);
    const [poster_path, setThumbnail] = useState('');
    const [title, setTitle] = useState('');
    const [overview, setDescription] = useState('');

    const [addRequestStatus, setAddRequestStatus] = useState('idle');
    const [isOpen, setIsOpen] = useState(false);
    const [modalTitle, setModalTitle] = useState('');
    const [modalContent, setModalContent] = useState('');
    const [modalIcon, setModalIcon] = useState(false);
    // RTKQ code
    // const [addNewData, { isLoading }] = useAddNewDataMutation()
    const dispatch = useAppDispatch();
    // RTKQ code
    // const canSave = [
    //     genre,
    //     url,
    //     rating,
    //     release_date,
    //     runtime,
    //     thumbnail,
    //     title,
    //     description,
    // ].every(Boolean) && !isLoading
    const canSave =
        [
            genres,
            vote_average,
            release_date,
            runtime,
            poster_path,
            title,
            overview,
        ].every(Boolean) && addRequestStatus === 'idle';

    const onGenreChanged = (e: [{ value: string; label: string }]) => {
        const genresData = e.map((genre) => genre.value);
        console.log(genresData);
        setGenre(genresData);
    };
    const onRatingChanged = (e: {
        target: { value: SetStateAction<string> };
    }) => {
        setRating(Number(e.target.value));
    };
    const onReleaseChanged = (e: { target: { value: string } }) => {
        const year = new Date(e.target.value).getFullYear().toString();
        setRelease(year);
    };
    const onRuntimeChanged = (e: {
        target: { value: SetStateAction<string> };
    }) => {
        setRuntime(Number(e.target.value));
    };
    const onThumbnailChanged = (e: {
        target: { value: SetStateAction<string> };
    }) => {
        setThumbnail(e.target.value);
    };
    const onTitleChanged = (e: {
        target: { value: SetStateAction<string> };
    }) => {
        setTitle(e.target.value);
    };
    const onDescriptionChanged = (e: {
        target: { value: SetStateAction<string> };
    }) => {
        setDescription(e.target.value);
    };

    const onSubmit = async () => {
        if (canSave) {
            try {
                setAddRequestStatus('pending');
                await dispatch(
                    addMovie({
                        title,
                        tagline: 'tagline',
                        vote_average,
                        vote_count: 0,
                        release_date,
                        poster_path,
                        overview,
                        budget: 0,
                        genres,
                        runtime,
                    })
                ).unwrap();
                setModalTitle('Congratulations!');
                setModalIcon(true)
                setModalContent(
                    `The movie has been added to database successfully`
                );
                setIsOpen(true);
                setGenre([]);
                setRating(0);
                setRelease('');
                setRuntime(0);
                setThumbnail('');
                setTitle('');
                setDescription('');
            } catch (err) {
                setModalTitle('Failure!');
                setModalContent(
                    `The movie has been failed to add to database :(`
                );
                setIsOpen(true);
            } finally {
                setAddRequestStatus('idle');
            }
            // RTKQ code
            // try {
            //     await addNewData({
            //         id: nanoid(),
            //         genre,
            //         url,
            //         rating,
            //         release_date,
            //         runtime,
            //         thumbnail,
            //         title,
            //         description,
            //     }).unwrap()
            // RTKQ code
            // } catch (error) {
            //     console.warn('Failed to add movie: ', error)
            // }
        } else {
            setModalTitle('Can\'t save movie');
            setModalContent(`Please, fill all the required fields`);
            setIsOpen(true);
        }
    };

    return (
        <>
            <form action=''>
                <h2 className={classes['movie-form__title']}>Add movie</h2>
                <div className={classes['movie-form__fieldset']}>
                    <div className={classes['movie-form__group']}>
                        <label htmlFor='movieTitle'>Title</label>
                        <input
                            type='text'
                            id='movieTitle'
                            name='movieTitle'
                            onChange={onTitleChanged}
                        />
                    </div>
                    <div className={classes['movie-form__group']}>
                        <label htmlFor='movieRelease'>Release date</label>
                        <input
                            type='date'
                            id='movieRelease'
                            name='movieRelease'
                            placeholder='Select Date'
                            onChange={onReleaseChanged}
                        />
                    </div>
                    <div className={classes['movie-form__group']}>
                        <label htmlFor='movieRating'>Rating</label>
                        <input
                            type='number'
                            id='movieRating'
                            name='movieRating'
                            min='1'
                            max='10'
                            step='0.1'
                            onChange={onRatingChanged}
                        />
                    </div>
                    <div className={classes['movie-form__group']}>
                        <label htmlFor='react-select-2-input'>Genre</label>
                        <CustomSelect
                            options={[
                                { value: 'crime', label: 'Crime' },
                                { value: 'documentary', label: 'Documentary' },
                                { value: 'horror', label: 'Horror' },
                                { value: 'comedy', label: 'Comedy' },
                                { value: 'fantasy', label: 'Fantasy' },
                            ]}
                            handleChange={onGenreChanged}
                        />
                    </div>
                    <div className={classes['movie-form__group']}>
                        <label htmlFor='movieRuntime'>Runtime</label>
                        <input
                            type='text'
                            id='movieRuntime'
                            name='movieRuntime'
                            onChange={onRuntimeChanged}
                        />
                    </div>
                    <div className={classes['movie-form__group']}>
                        <label htmlFor='movieThumbnail'>Poster image URL</label>
                        <input
                            type='text'
                            id='movieThumbnail'
                            name='movieThumbnail'
                            onChange={onThumbnailChanged}
                        />
                    </div>
                    <div
                        className={`${classes['movie-form__group']} ${classes['movie-form__group--full']}`}
                    >
                        <label htmlFor='movieOverview'>Overview</label>
                        <textarea
                            id='movieOverview'
                            name='movieOverview'
                            rows={3}
                            onChange={onDescriptionChanged}
                        ></textarea>
                    </div>
                    <div className={classes['movie-form__actions']}>
                        <Button isGhost={true}>reset</Button>
                        <Button type='button' onClick={onSubmit}>
                            submit
                        </Button>
                    </div>
                </div>
            </form>
            <Modal
                handleClose={() => setIsOpen(false)}
                isOpen={isOpen}
                size={'medium'}
                title={modalTitle}
            >
                {modalIcon ? 
                    <img
                        src={IconCheck}
                        alt=''
                        className={modalClasses['modal__icon']}
                    /> : ''
                }
                <p>{modalContent}</p>
            </Modal>
        </>
    );
};

export default AddMovieForm;
