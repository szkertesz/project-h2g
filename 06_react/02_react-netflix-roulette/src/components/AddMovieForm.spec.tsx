import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import AddMovieForm from './AddMovieForm';
import { addMovie } from '../features/movies/moviesSlice';

const mockStore = configureStore([]);

describe('AddMovieForm', () => {
    let store;
    let component: { toJSON: () => any };

    beforeEach(() => {
        store = mockStore({
            initialState: {
                data: [],
                status: 'idle',
                error: null,
                searchedMovie: '',
                sortOptions: {
                    sortCriterion: '',
                    sortOrder: '',
                },
                filterOptions: {
                    genre: '',
                },
            },
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <AddMovieForm />
            </Provider>
        );
    });
    // the test suite
    it('should render AddMovieForm component', () => {
        expect(component.toJSON()).toMatchSnapshot(); // the assertion
    });
});
