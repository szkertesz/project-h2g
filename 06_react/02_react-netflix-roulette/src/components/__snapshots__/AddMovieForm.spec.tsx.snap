// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieForm should render AddMovieForm component 1`] = `
<form
  action=""
>
  <h2
    className="movie-form__title"
  >
    Add movie
  </h2>
  <div
    className="movie-form__fieldset"
  >
    <div
      className="movie-form__group"
    >
      <label
        htmlFor="movieTitle"
      >
        Title
      </label>
      <input
        id="movieTitle"
        name="movieTitle"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="movie-form__group"
    >
      <label
        htmlFor="movieRelease"
      >
        Release date
      </label>
      <input
        id="movieRelease"
        name="movieRelease"
        onChange={[Function]}
        placeholder="Select Date"
        type="date"
      />
    </div>
    <div
      className="movie-form__group"
    >
      <label
        htmlFor="movieRating"
      >
        Rating
      </label>
      <input
        id="movieRating"
        max="10"
        min="1"
        name="movieRating"
        onChange={[Function]}
        step="0.1"
        type="number"
      />
    </div>
    <div
      className="movie-form__group"
    >
      <label
        htmlFor="react-select-2-input"
      >
        Genre
      </label>
      <div
        className=" css-1plhcwh-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-on302j-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-k4907d-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-2-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-14msk06-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <img
                alt=""
                src="icon-arrow.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-form__group"
    >
      <label
        htmlFor="movieRuntime"
      >
        Runtime
      </label>
      <input
        id="movieRuntime"
        name="movieRuntime"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="movie-form__group"
    >
      <label
        htmlFor="movieThumbnail"
      >
        Poster image URL
      </label>
      <input
        id="movieThumbnail"
        name="movieThumbnail"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="movie-form__group movie-form__group--full"
    >
      <label
        htmlFor="movieOverview"
      >
        Overview
      </label>
      <textarea
        id="movieOverview"
        name="movieOverview"
        onChange={[Function]}
        rows={3}
      />
    </div>
    <div
      className="movie-form__actions"
    >
      <button
        className="button 
            button--ghost
            "
      >
        reset
      </button>
      <button
        className="button 
            button--default
            "
        onClick={[Function]}
        type="button"
      >
        submit
      </button>
    </div>
  </div>
</form>
`;
